openedx/core/djangoapps/theming/template_loaders.py

```diff
@@ -41,7 +41,7 @@ def get_template_sources(self, template_name, template_dirs=None):
         if isinstance(theme_dirs, list):
             template_dirs = theme_dirs + template_dirs
 
-        return list(super(ThemeFilesystemLoader, self).get_template_sources(template_name, template_dirs))
+        return list(super(ThemeFilesystemLoader, self).get_template_sources(template_name))
```

Comment:
This change was made to the Django template loader API because the templates directory is now specified in the template engine initialization rather than every call to load a template.  But whereas the loaders above in `makoloader.py` all delegate to stock Django loaders which are associated with an engine that already has the directories configured, this one is making changes to the directory list just before the template loading.  I think to avoid theming regressions, we may need a second engine with the above changes to `template_dirs` applied on initialization, that is only used for this loader.  We'd then need to update the `TEMPLATES` setting in both LMS and Studio to configure each engine and its loaders correctly.

Let me know if you're comfortable attempting this or would rather someone in the Cambridge office try to sort it out.

Reply:
sure, as I'm already working on it, I'll go ahead and add a new loader that is compatible with these changes.